core/trino-main/src/main/java/io/trino/execution/scheduler/ScaledWriterScheduler.java

```diff
@@ -100,10 +97,15 @@ private int getNewTaskCount()
                 .mapToLong(DataSize::toBytes)
                 .sum();
 
+        long minWrittenBytesToScaleUp = writerTasksProvider.get().stream()
```

Comment:
Multiplication by `scheduledNodes.size()` is intentionally missing?

Reply:
It's the same logic as before just that we are now using task level writer count instead of a common value. So, what I mean is basically `scheduledNodes.size()` == `writerTasksProvider.get().size()`